packages/flutter/lib/src/cupertino/adaptive_text_selection_toolbar.dart

```diff
@@ -0,0 +1,375 @@
+import 'desktop_text_selection_toolbar_button.dart';
+import 'localizations.dart';
+import 'text_selection_toolbar.dart';
+import 'text_selection_toolbar_button.dart';
+
+/// The default Cupertino context menu for text selection for the current
+/// platform with the given children.
+///
+/// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.platforms}
+/// Builds the mobile Cupertino context menu on all mobile platforms, not just
+/// iOS, and builds the desktop Cupertino context menu on all desktop platforms,
+/// not just MacOS. For a widget that builds the native-looking context menu for
+/// all platforms, see [AdaptiveTextSelectionToolbar].
+/// {@endtemplate}
+///
+/// See also:
+///
+/// * [getEditableTextButtonItems], which generates the default
+///   [ContextMenuButtonItem]s for a context menu displaying inside of an
+///   [EditableText] on the current platform.
+/// * [AdaptiveTextSelectionToolbar], which does the same thing as this widget
+///   but for all platforms, not just the Cupertino-styled platforms.
+/// * [CupertinoAdaptiveTextSelectionToolbar.getAdaptiveButtons], which builds
+///   the Cupertino button Widgets for the current platform given
+///   [ContextMenuButtonItem]s.
+class CupertinoAdaptiveTextSelectionToolbar extends StatelessWidget {
+  /// Create an instance of [CupertinoAdaptiveTextSelectionToolbar] with the
+  /// given [children].
+  ///
+  /// See also:
+  ///
+  /// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.buttonItems}
+  /// * [CupertinoAdaptiveTextSelectionToolbar.buttonItems], which takes a list
+  ///   of [ContextMenuButtonItem]s instead of [children] widgets.
+  /// {@endtemplate}
+  /// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.editable}
+  /// * [CupertinoAdaptiveTextSelectionToolbar.editable], which builds the
+  ///   default Cupertino children for an editable field.
+  /// {@endtemplate}
+  /// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.editableText}
+  /// * [CupertinoAdaptiveTextSelectionToolbar.editableText], which builds the
+  ///   default Cupertino children for an [EditableText].
+  /// {@endtemplate}
+  /// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.selectable}
+  /// * [CupertinoAdaptiveTextSelectionToolbar.selectable], which builds the
+  ///   Cupertino children for content that is selectable but not editable.
+  /// {@endtemplate}
+  /// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.selectableRegion}
+  /// * [CupertinoAdaptiveTextSelectionToolbar.selectableRegion], which builds
+  ///   the default Cupertino children for a [SelectableRegion].
+  /// {@endtemplate}
+  const CupertinoAdaptiveTextSelectionToolbar({
+    super.key,
+    required this.children,
+    required this.primaryAnchor,
+    this.secondaryAnchor,
+    final TargetPlatform? targetPlatform,
+  }) : buttonItems = null,
+       _targetPlatform = targetPlatform;
```

Comment:
Out of curiosity, is there a reason to provide `targetPlatform` as an explicit parameter instead of using `Theme.of(context).targetPlatform`? Even if there's no `Theme` in the context (e.g. non-Material app?), it should default to `ThemeData.fallback`.

Reply:
Yes the reason is to keep Cupertino free of stuff from the Material library.  You're right though, if I just remove this parameter everywhere the result is the same as using defaultTargetPlatform.  I'll do that.